backend/src/bundles/google-fit/helpers/prepare-actions.helper.ts

```diff
@@ -0,0 +1,69 @@
+import { Action } from '../enums/enums.js';
+import {
+    type PrepareActionsResponse,
+    type WorkoutRequestDto,
+} from '../types/types.js';
+
+const isEquivalentWorkout = (
+    a: WorkoutRequestDto,
+    b: WorkoutRequestDto,
+): boolean => {
+    const fieldsToCompare: (keyof WorkoutRequestDto)[] = [
+        'activityType',
+        'speed',
+        'distance',
+        'heartRate',
+        'activityId',
+        'kilocalories',
+    ];
+
+    for (const field of fieldsToCompare) {
+        if (a[field] !== b[field]) {
+            return false;
+        }
+    }
+
+    if ((a.steps || 0) !== (b.steps || 0)) {
+        return false;
+    }
+
+    return (
+        a.workoutStartedAt.toISOString() === b.workoutStartedAt.toISOString() &&
+        a.workoutEndedAt.toISOString() === b.workoutEndedAt.toISOString()
+    );
+};
+
+const prepareActions = (
```

Comment:
I think we could just get two sets of activity ids from DB and from response and find 
- Differences for create / delete
- Intersection for updates
Then map results into actions, should be a lot more efficient